sha1/src/checked/ubc_check.rs

```diff
@@ -0,0 +1,1179 @@
+//! Direct translation of the C code
+//! Original license and comments:
+//!
+//! Copyright 2017 Marc Stevens <marc@marc-stevens.nl>, Dan Shumow <danshu@microsoft.com>
+//! Distributed under the MIT Software License.
+//! See accompanying file LICENSE.txt or copy at
+//! https://opensource.org/licenses/MIT
```

Comment:
This code is MIT only which slightly changes the licensing situation for this crate, which is otherwise Apache 2.0+MIT.

We could note that the code for the `collision` feature is MIT-only in the toplevel README.md, or potentially ask the original authors if they would be okay with the resulting translation being dual licensed as Apache 2.0 in addition to MIT.

Reply:
also @shumow

Reply:
Hi Friedel, Tony, 
Thanks for your nice work making our code also available in rust!
And I'm certainly fine with dual licensing Apache 2.0+MIT (with the proper attribution and copyright notices). 
But wouldn't the same situation also apply to the rust translation of our `sha1.c` code?

Reply:
> And I'm certainly fine with dual licensing Apache 2.0+MIT (with the proper attribution and copyright notices).

awesome, thank you!

> But wouldn't the same situation also apply to the rust translation of our sha1.c code?

Yes very much, I will be doing a final pass of notes for attribution on the relevant files in a bit. I'll ping you when it is ready for review.

Reply:
I have added a detailed reference to `src/checked.rs` and pointers to this and the original files in both `ubc_check.rs` and `compress/checked.rs`. Please let me know if this is good for you